ndd_slurm.py

```diff
@@ -67,15 +69,15 @@ def get_slave_parser():
 
 def get_master_input_args(args, read_fd):
```

Comment:
You may change this parameters to something like `input`:
```
def get_master_input_args(input):
  return ['-i', input] if input or ['-I', '/dev/stdin']
```
And then call it like `get_master_input_args(None if read_fd or args.i)`

This way it would have less dependencies and look simpler.